hw/opentitan/ot_spi_host.c

```
@@ -1293,6 +1291,66 @@ static void ot_spi_host_io_write(void *opaque, hwaddr addr, uint64_t val64,
     }
 }
 
+static uint8_t ot_spi_host_downstream_transfer(OtSPIHostState *s, uint8_t tx)
+{
+    /* Passthrough is not implemented if SPI Host has more than one CS line */
+    if (s->num_cs != 1u) {
+        trace_ot_spi_host_passthrough_unimplemented(s->ot_id);
+        return SPI_DEFAULT_TX_VALUE;
+    }
+
+    if (!s->passthrough_en) {
+        trace_ot_spi_host_passthrough_disabled(s->ot_id);
+        return SPI_DEFAULT_TX_VALUE;
+    }
+
+    /* Forward to downstream flash */
+    return (uint8_t)ssi_transfer(s->ssi, (uint32_t)tx);
+}
+
+static void
+ot_spi_host_device_passthrough_en_input(void *opaque, int irq, int level)
+{
+    OtSPIHostState *s = opaque;
+    g_assert(irq == 0u);
+
+    /* Passthrough is not implemented if SPI Host has more than one CS line */
+    if (s->num_cs != 1u) {
+        trace_ot_spi_host_passthrough_unimplemented(s->ot_id);
+        return;
+    }
+
+    s->passthrough_en = (bool)level;
+
+    /*
+     * Real HW muxes the CS line based on Passthrough Enable, but it doesn't
+     * make sense for a transfer to be handled by both SPI Host and SPI Device,
```

Some `LOG_GUEST_ERROR` is definitely **needed** here:
It does not make sense, but it can happen (so it will :-)). It's quite trivial to have two unrelated tasks that try to access the same HW resource.
If the HW does not manage exclusive access to the SPI bus (which is really weird but that's out of the scope), QEMU should neither do it, but it should inform the user that something unexpected is going on.

A similar `LOG_GUEST_ERROR` message should be emitted when the SPI host is trying to access the bus while 
`passthrough_en` is enabled.

---

The means of exclusive access to the SPI bus is the Passthrough Enable signal, which controls the mux. I thought it would be really pathological for something to start on the SPI Host, have SPI device assert chip select too, then enable Passthrough so that it is controlling the bus for the remainder of the same transfer, so that is what I aimed to try and prevent. Do you mean that I should relax this and add a `GUEST_ERROR` warning if changing the Passthrough Enable when both have a Chip Select active?

---

I'm not sure how to define pathological here, but a simple SW bug would be enough. 

It is a somewhat weird design to have one HW device silently using another HW device with no way to control this from SW (_i.e._ through a register in SPI host that manages the mux). That's what pinmuxes are for (among other things)...

What I meant is that if SPI device try to set `passthrough_en` while SPI host is using the same bus (i.e. it has asserted the /CS line), a LOG_GUEST_ERROR should be emitted, and it should take priority over SPI host, potentially trashing the flash.
 
Conversely, if the SPI host is trying to assert /CS while `passthough_en` is enabled, a `LOG_GUEST_ERROR` should be emitted. Moreover, in this case, its /CS action or `ssi_transfer` requests should be discarded.

---

I now track the passthrough chip select, passthrough enable, and chip select from the SPI Host, and mux those faithfully onto the CS[0] IRQ. I've added guest errors around any suspicious transitions.